### PR TITLE
Simplify RPC command registration

### PR DESCRIPTION
**Pull request by awemany**
Also get rid of the premature optimization that is avoiding copies
of the CRPCCommand structs when registering the RPCs.

### CODE REVIEW

**Comment by awemany** [comment]
Putting into WIP, will do another couple simplifications.

File: src/rpc/server.cpp [hunk 1]
```diff
@@ -255,35 +256,29 @@ static const CRPCCommand vRPCCommands[] = {
...
+    for (auto cmd : vRPCCommands)
+        appendCommand(cmd);
 }
 
 const CRPCCommand *CRPCTable::operator[](const std::string &name) const
 {
-    map<string, const CRPCCommand *>::const_iterator it = mapCommands.find(name);
+    map<string, CRPCCommand>::const_iterator it = mapCommands.find(name);
     if (it == mapCommands.end())
         return NULL;
```
Comment (sickpig) [inline src/rpc/server.cpp]: while at it why not using `nullptr` here?

File: src/test/rpc_tests.cpp [hunk 2]
```diff
@@ -406,4 +406,12 @@ BOOST_AUTO_TEST_CASE(rpc_convert_values_generatetoaddress)
...
 }
 
+BOOST_AUTO_TEST_CASE(rpc_help)
+{
+    const string s = tableRPC.help("");
+    // check sorting by category (exactly one entry named 'Mining')
+    size_t p = s.find("== Mining ==");
+    BOOST_CHECK(p != string::npos);
+    BOOST_CHECK(s.substr(p + 1).find("== Mining ==") == string::npos);
+}
```
Comment (sickpig) [inline src/test/rpc_tests.cpp:416]: :+1:

**Review by sickpig — APPROVED** [review]
approved a part from very minor nit

File: src/rpc/server.cpp (hunk 1 above)
Comment (awemany) [reply src/rpc/server.cpp]: Good point. Changed. I also replaced another `pair<...>` that I forgot with the `rpc_pair` typedef.

**Comment by gandrewstone** [comment]
Given the quantity of RAM available to modern machines, avoiding duplication of the RPC table seems irrelevant.

**Pull request merged**